src/test/java/datawave/query/composite/CompositeMetadataTest.java

```diff
@@ -79,4 +84,27 @@ public void readWriteCompositeMetadataTest() {
         }
     }
     
+    @Test
+    public void serializeManyThreads() {
+        final ExecutorService executor = Executors.newFixedThreadPool(5);
+        try {
+            for (int i = 0; i < 100; i++) {
+                executor.submit(() -> {
+                    byte[] compMetadataBytes = CompositeMetadata.toBytes(compositeMetadata);
+                    Assert.assertNotNull(compMetadataBytes);
+                });
+            }
+        } finally {
+            // added protection
+            executor.shutdown();
+            try {
+                // should not need this, but will perform for safety sake
+                executor.awaitTermination(10, TimeUnit.SECONDS);
```

Comment:
10 seconds is arbitrary. happy to make higher/lower...but given the simplicity of the operations, we should never hit ten seconds.